mne/source_space.py

```diff
@@ -1549,19 +1549,28 @@ def _make_voxel_ras_trans(move, ras, voxel_size):
     return t
 
 
-def _make_discrete_source_space(pos):
+def _make_discrete_source_space(pos, coord_frame='mri'):
     """Use a discrete set of source locs/oris to make src space
 
     Parameters
     ----------
     pos : dict
         Must have entries "rr" and "nn". Data should be in meters.
+    coord_frame : str
+        The coordinate frame in which the positions are given; default: 'mri'.
+        The frame must be one defined in transforms.py:_str_to_frame
 
     Returns
     -------
     src : dict
         The source space.
     """
+    # Check that coordinate frame is valid
+    if coord_frame not in _str_to_frame:  # will fail if coord_frame not string
+        raise KeyError('coord_frame must be one of %s, not "%s"'
+                       % (list(_str_to_frame.keys()), coord_frame))
```

Comment:
we should refactor `_check_frame` from `mne/channels/montage.py`, and move it to `mne/transforms.py` in the process